tests/test_main.py

```diff
@@ -176,9 +177,20 @@ def test_no_child_at_all(self):
         self.assertIsNone(child)
 
     def test_normal_child_program(self):
-        analyzable, child = main.decide_child_program(False, "foobar.py")
-        self.assertEqual(analyzable, "foobar.py")
-        self.assertEqual(child, "foobar.py")
+        child_path = create_tempfile(self, "")
+        analyzable, child = main.decide_child_program(False, child_path)
+        self.assertEqual(analyzable, child_path)
+        self.assertEqual(child, child_path)
+
+    def test_normal_child_program_not_found(self):
+        with self.assertRaises(FadesError):
+            main.decide_child_program(False, 'does_not_exist.py')
+
+    def test_normal_child_program_no_access(self):
+        child_path = create_tempfile(self, "")
+        os.chmod(child_path, 333)  # Remove read permission.
```

Comment:
Probably you should change the permission back to something "sane" after the test (make sure it's before the actual file removal from create_tempfile)